src/effects/specialeffectchainslots.cpp

```diff
@@ -0,0 +1,197 @@
+    const QSet<ChannelHandleAndGroup>& registeredChannels =
+            m_pEffectsManager->registeredInputChannels();
+    for (const ChannelHandleAndGroup& handle_group : registeredChannels) {
+        if (handle_group.name() == "[MasterOutput]") {
+            masterHandleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(masterHandleAndGroup != nullptr);
+
+    registerInputChannel(*masterHandleAndGroup);
+    enableForInputChannel(*masterHandleAndGroup);
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+}
+
+QString OutputEffectChainSlot::formatEffectChainSlotGroup(const QString& group) {
+    return QString("[OutputEffectRack_%1]").arg(group);
+}
+
+
+PerGroupEffectChainSlot::PerGroupEffectChainSlot(const QString& group,
+                                                 const QString& chainSlotGroup,
+                                                 unsigned int iChainNumber,
+                                                 EffectsManager* pEffectsManager)
+        : EffectChainSlot(chainSlotGroup, iChainNumber, pEffectsManager,
+                          SignalProcessingStage::Prefader, false,
+                          chainSlotGroup) {
+    // Set the chain to be fully wet.
+    m_pControlChainMix->set(1.0);
+    sendParameterUpdate();
+
+    // TODO(rryan): remove.
+    const ChannelHandleAndGroup* handleAndGroup = nullptr;
+    for (const ChannelHandleAndGroup& handle_group :
+             m_pEffectsManager->registeredInputChannels()) {
+        if (handle_group.name() == group) {
+            handleAndGroup = &handle_group;
+            break;
+        }
+    }
+    DEBUG_ASSERT(handleAndGroup != nullptr);
+
+    // Register this channel alone with the chain slot.
+    registerInputChannel(*handleAndGroup);
+    enableForInputChannel(*handleAndGroup);
+
+    // Add a single effect slot
+    addEffectSlot(chainSlotGroup);
+    // DlgPrefEq loads the Effect with loadEffectToGroup
+}
+
+
+QuickEffectChainSlot::QuickEffectChainSlot(const QString& group,
+                                           unsigned int iChainNumber,
+                                           EffectsManager* pEffectsManager)
+        : PerGroupEffectChainSlot(group, formatEffectChainSlotGroup(group),
+                                  iChainNumber, pEffectsManager) {
+    configureForGroup(group);
+}
+
+QString QuickEffectChainSlot::formatEffectChainSlotGroup(const QString& group) {
+    return formatEffectChainSlotGroupString(group);
+}
+
+QString QuickEffectChainSlot::formatEffectChainSlotGroupString(const QString& group) {
+    return QString("[QuickEffectRack1_%1]").arg(group);
+}
+
+void QuickEffectChainSlot::configureForGroup(const QString& group) {
+    // Set the parameter default value to 0.5 (neutral).
+    setSuperParameter(0.5);
+    setSuperParameterDefaultValue(0.5);
+}
+
+QString QuickEffectChainSlot::formatEffectSlotGroup(const QString& group) {
```

Comment:
This assumes that there will only ever be one effect in each QuickEffect chain. After we implement saving/loading custom chains, I would like it to be possible to load a custom chain preset for the QuickEffects, so that assumption would get in the way.